app/views/NotificationPreferencesView/index.tsx

```diff
@@ -1,282 +1,234 @@
+
+		const options: TActionSheetOptionsItem[] = OPTIONS[preference].map(i => ({
+			title: I18n.t(i.label, { defaultValue: i.label, second: i.second }),
+			onPress: () => {
+				hideActionSheet();
+				onChangeValue(preference, { [preference]: i.value.toString() }, () => setOption(option));
+				setOption(i);
+			},
+			right: option?.value === i.value ? () => <CustomIcon name={'check'} size={20} color={colors.tintActive} /> : undefined
+		}));
 
-	componentWillUnmount() {
-		if (this.subscription && this.subscription.unsubscribe) {
-			this.subscription.unsubscribe();
-		}
+		return (
+			<List.Item
+				title={title}
+				testID={testID}
+				onPress={() => showActionSheet({ options })}
+				right={() => (
+					<Text style={[{ ...sharedStyles.textRegular, fontSize: 16 }, { color: colors.actionTintColor }]}>
+						{option?.label ? I18n.t(option?.label, { defaultValue: option?.label, second: option?.second }) : option?.label}
+					</Text>
+				)}
+			/>
+		);
 	}
+);
+
+const RenderSwitch = ({ preference, room, onChangeValue }: IBaseParams) => {
+	const [switchValue, setSwitchValue] = useState(!room[preference]);
+	return (
+		<Switch
+			value={switchValue}
+			testID={preference}
+			trackColor={SWITCH_TRACK_COLOR}
+			onValueChange={value => {
+				onChangeValue(preference, { [preference]: switchValue ? '1' : '0' }, () => setSwitchValue(switchValue));
+				setSwitchValue(value);
+			}}
+		/>
+	);
+};
+
+const NotificationPreferencesView = () => {
+	const navigation = useNavigation();
+	const route = useRoute<RouteProp<ChatsStackParamList, 'NotificationPrefView'>>();
+	const serverVersion = useAppSelector(state => state.server.version);
+
+	useEffect(() => {
+		navigation.setOptions({
+			title: I18n.t('Notification_Preferences')
+		});
+	}, []);
 
-	saveNotificationSettings = async (key: string, value: string | boolean, params: IRoomNotifications) => {
-		// @ts-ignore
-		logEvent(events[`NP_${key.toUpperCase()}`]);
-		const { room } = this.state;
-		const db = database.active;
+	const rid = route.params?.rid ?? '';
+	const room = route.params?.room;
 
+	const saveNotificationSettings = async (key: string, params: IRoomNotifications, onError: Function) => {
 		try {
-			await db.write(async () => {
-				await room.update(
-					protectedFunction((r: IRoomNotifications) => {
-						r[key] = value;
-					})
-				);
-			});
-
-			try {
-				const result = await Services.saveNotificationSettings(this.rid, params);
-				if (result.success) {
-					return;
-				}
-			} catch {
-				// do nothing
-			}
-
-			await db.write(async () => {
-				await room.update(
-					protectedFunction((r: IRoomNotifications) => {
-						r[key] = room[key];
-					})
-				);
-			});
+			// @ts-ignore
+			logEvent(events[`NP_${key.toUpperCase()}`]);
+			await Services.saveNotificationSettings(rid, params);
 		} catch (e) {
 			// @ts-ignore
 			logEvent(events[`NP_${key.toUpperCase()}_F`]);
 			log(e);
+			onError();
+			showErrorAlertWithEMessage(e);
```

Comment:
This is confusing.
When is it calling `onError`?
Will it do both `onError` and `showErrorAlertWithEMessage`?